### PR TITLE
Add notebooks for ORT training: summarization and text classification

### PR DESCRIPTION
**Pull request by JingyaHuang**
# What does this PR do?

## Content

* Add a notebook on the text classification task for `ORTTrainer`
* Add a notebook on the summarization task for `ORTSeq2SeqTrainer`

### CODE REVIEW

**Comment by JingyaHuang** [comment]
Not sure how it works for the notebook repo. Gently pinging team members for review. 
@echarlaix @regisss @michaelbenayoun @fxmarty @mht-sharma

File: examples/text_classification-ort.ipynb [hunk 1]
```diff
@@ -0,0 +1,1289 @@
...
+   "source": [
+    "!nvidia-smi"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e379d8ab-9495-4c2a-ace9-f1e605ea9adf",
+   "metadata": {},
+   "source": [
+    "If you're opening this Notebook on colab, you will probably need to install 🤗 Optimum, 🤗 Transformers, 🤗 Datasets and 🤗 evaluate. Uncomment the following cell and run it."
```
Comment (fxmarty) [inline examples/text_classification-ort.ipynb:68]: Probably we should hint to install `optimum[onnnxruntime-gpu]`? And also `torch_ort`?

File: examples/text_classification-ort.ipynb [hunk 2]
```diff
@@ -0,0 +1,1289 @@
...
+  {
+   "cell_type": "markdown",
+   "id": "a9408e05-175b-4573-8314-9ae4ba7d3453",
+   "metadata": {},
+   "source": [
+    "__Dependencies__\n",
+    "\n",
+    "To use ONNX Runtime for training, you need a machine with at least one NVIDIA or AMD GPU.\n",
+    "\n",
+    "CUDA and cuDNN need to be properly installed, before the installation of ONNX Runtime training module. Please follow the instruction in [Optimum's documentation](https://huggingface.co/docs/optimum/onnxruntime/trainer) to set up your environment.\n",
```
Comment (fxmarty) [inline examples/text_classification-ort.ipynb:24]: https://huggingface.co/docs/optimum/onnxruntime/trainer currently does not give info on how to setup the environment.

**Comment by fxmarty** [comment]
That's cool @JingyaHuang ! Do you think it would make sense to have a section comparing e.g. how long an epoch takes with `Trainer` vs `ORTTrainer`?

File: examples/text_classification-ort.ipynb (hunk 2 above)
Comment (JingyaHuang) [reply examples/text_classification-ort.ipynb:24]: Hi @fxmarty,

That's true, the documentation with detailed setup steps will be updated in [PR#376](https://moon-ci-docs.huggingface.co/docs/optimum/pr_376/en/onnxruntime/trainer#prerequisite). I temporarily put the link to the main doc, this shall be replaced after the doc makes its way to the main of optimum, thanks for noticing.

File: examples/text_classification-ort.ipynb (hunk 1 above)
Comment (JingyaHuang) [reply examples/text_classification-ort.ipynb:68]: I actually won't recommend users to set up the prerequisite(including onnx runtime dependencies and torch_ort) manually in the notebook, I prefer to let them set it up first either locally or with the docker files that we offer, and then install optimum and transformers, etc. in the notebook. I will make it more clear in the document that they need to set up the env first.

**Comment by JingyaHuang** [comment]
> That's cool @JingyaHuang ! Do you think it would make sense to have a section comparing e.g. how long an epoch takes with `Trainer` vs `ORTTrainer`?

I once did a rough benchmark and we will communicate with ONNX Runtime for more stats, but I think of putting it in the readme of optimum examples. Notebook is more la step-by-step tutorial for me, WDYT?

File: examples/text_classification-ort.ipynb (hunk 2 above)
Comment (fxmarty) [reply examples/text_classification-ort.ipynb:24]: Ok great I missed it!

**Comment by fxmarty** [comment]
Ok great, yes I think it's good if there are some stats available to motivate people to use it.

Also, how are the notebooks usually advertised? Would it make sense to include (part of) it in e.g. a tutorial section in the Optimum doc, a bit like the how-to guides in transformers doc?

File: examples/summarization-ort.ipynb [hunk 3]
```diff
@@ -0,0 +1,1443 @@
...
+  {
+   "cell_type": "markdown",
+   "id": "6f5e9a38-156d-45ce-8ddc-c06c76bef057",
+   "metadata": {},
+   "source": [
+    "## __Extended reading__\n",
+    "\n",
+    "Now check your trained ONNX model with Netron, and you might notice that the computation graph is under optimizatiom. Want to accelerate even more? \n",
+    "\n",
+    "Check the graph optimizers and quantizers of [Optimum](https://github.com/huggingface/optimum)🤗! "
```
Comment (echarlaix) [inline examples/summarization-ort.ipynb:1410]:
```suggestion
    "Check the graph [optimizers](https://huggingface.co/docs/optimum/onnxruntime/optimization) and [quantizers](https://huggingface.co/docs/optimum/onnxruntime/quantization) of [Optimum](https://github.com/huggingface/optimum)🤗! "
```

File: examples/text_classification-ort.ipynb [hunk 4]
```diff
@@ -0,0 +1,1289 @@
...
+    "name": "ipython",
+    "version": 3
+   },
+   "file_extension": ".py",
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython3",
+   "version": "3.8.10"
+  }
```
Comment (echarlaix) [inline examples/text_classification-ort.ipynb:1285]: I would rename the file `text_classification_ort.ipynb` to keep consistency with the existing ort examples

**Review by echarlaix — APPROVED** [review]
LGTM, thanks for adding theses examples @JingyaHuang !

**Comment by JingyaHuang** [comment]
> Ok great, yes I think it's good if there are some stats available (not necessarily in the notebooks) to motivate people to use it.
> 
> Also, how are the notebooks usually advertised? Would it make sense to include (part of) it in e.g. a tutorial section in the Optimum doc, a bit like the how-to guides in transformers doc?

I will open an PR in [Optimum notebooks](https://github.com/huggingface/optimum/tree/main/notebook) once the notebooks are merged. And probably also in [transformers' notebook catalog](https://github.com/huggingface/transformers/tree/main/notebooks).

**Pull request merged**